### ISSUE TITLE
[Feature] Add an opt-in referral code

### ISSUE THREAD

**Issue by qtdyn** [Abdallah-Alwarawreh/Syrup#3]
Thanks for the nice project, I hope to see it grow more :)

Many people seem to be fine with a money saving extension using its own coupon code, as long as it's either transparent, and/or doesn't override already existing referral codes.

I'd say adding an opt-in toggle in some settings to add said feature would be pretty neat, and would be rewarding you for your work, or simply to allow you to host the extension.

Thoughts?

**Comment by Abdallah-Alwarawreh** [comment]
Thanks!

Definitely! I'm planning on doing opt in settings, hopefully i can get a new version out in the next days with better design, settings and more quality of life features.

**Comment by Abdallah-Alwarawreh** [comment]
Checking back on this, I don't think I could do the opt in for applying my affiliate link because to do so I have to add the permission to read cookies, which can lead to mistrust and manual config for each site since I don't want to override other's affiliate links.

For now i will rely on ads on the [Syrup website](https://joinsyrup.com) (if you use the extension you wont get any ads)

If you have any ideas let me know!

**Comment by Azyrod** [comment]
I think it is expected for such an extension to work like this : 
## Affiliate links: 
  - If user already has affiliate link -> dont replace it. 
  - If not, put your own (doesn't harm anyone, and allows you to pay for hosting. You can have that opt-in but I don't see an issue not having it opt-in). 

This logic runs regardless of coupons logic.

## Coupons:
  - If user has coupon code but you know a better one, prompt the user if he'd like to switch coupons (show both coupons value and let him decide if he wants to support the creator who gave him the lower coupon, or use the better coupon promoted by your extension)
  - If user doesn't have a coupon code, automatically inject the best one and tell the user about it (user can still manually remove the coupon and replace it with a different one if he so wishes). 
  - If no coupons where found, do nothing (no useless "no coupons found" popup)


This guarantees the user can maximize his savings because you check what coupon he is already using and see if you have a better one. As a result of providing value to the customer, I think you are entitled to put your own affiliate link if no other one is present.

I would even add that, in case you find a better coupon and there is already an affiliate link, you can prompt the user if he whishes to swap the affiliate links as well since you are the source of the best coupon (which could be an opt-in option to do by default or not). He can also choose not to, and reward the creator who directed him to the product originally with the affiliate link, while using a different creator's coupon that you found for him.

Basically, as long as the user is in control of the behavior of the app, and nothing is done behind his back, it is fine.

**Comment by ImGajeed76** [comment]
@Azyrod 
I understand the suggestion about handling affiliate links. However, automatically inserting our own link when none is present (even as an opt-out feature) raises a fairness issue for the stores. Affiliate commissions are intended to reward promotion that drives customers. Inserting our link without performing that promotion means the store pays us for a customer they likely would have had anyway, which undermines the purpose of the affiliate program.

**Comment by Azyrod** [comment]
That's a fair point. Then you can only insert your link when you do find a better coupon, and if another link is present either leave it or prompt the user to choose which one to keep

I do think you are entitled to get something out of the value you are providing, even if it's only to pay for hosting
